pallets/domains/src/lib.rs

```diff
@@ -350,6 +403,41 @@ pub mod pallet {
             Self::deposit_event(Event::NewTldsSupported { count: inserted_tlds_count });
             Ok(Some(<T as Config>::WeightInfo::support_tlds(inserted_tlds_count)).into())
         }
+
+        #[pallet::call_index(7)]
+        #[pallet::weight((
+            T::DbWeight::get().writes(1) + Weight::from_ref_time(100_000),
+            DispatchClass::Operational,
+            Pays::Yes,
+        ))]
+        pub fn set_payment_beneficiary(
+            origin: OriginFor<T>,
+            payment_beneficiary: T::AccountId,
+        ) -> DispatchResultWithPostInfo {
+            ensure_root(origin)?;
+            PaymentBeneficiary::<T>::set(payment_beneficiary);
+            Ok(Pays::No.into())
+        }
+
+        #[pallet::call_index(8)]
+        #[pallet::weight((
+            T::DbWeight::get().writes(1) +
+                Weight::from_ref_time(100_000 * new_prices_config.len() as u64 * 2),
+            DispatchClass::Operational,
+            Pays::Yes,
+        ))]
+        pub fn set_price_config(
+            origin: OriginFor<T>,
+            mut new_prices_config: PricesConfigVec<T>,
+        ) -> DispatchResultWithPostInfo {
+            ensure_root(origin)?;
+
+            new_prices_config.sort_by_key(|(length, _)| *length);
+            new_prices_config.dedup_by_key(|(length, _)| *length);
+
+            PricesConfig::<T>::set(new_prices_config);
```

Comment:
Extract to `fn do_set_prices_config()` and reuse in the `GenesisBuild` instead of `Pallet::<T>::init_pallet(...)`.